langchain/src/tools/gmail/sent_message.ts

```diff
@@ -0,0 +1,88 @@
+import { google } from "googleapis";
+import { GmailBaseTool, GmailBaseToolParams } from "./base.js";
+
+interface SendMessageParams {
+  message: string;
+  to: string | string[];
+  subject: string;
+  cc?: string | string[];
+  bcc?: string | string[];
+}
+
+export class GmailSendMessage extends GmailBaseTool {
+  name = "gmail_send_message";
+
+  description = "Send a message using Gmail";
+
+  constructor(fields: GmailBaseToolParams) {
+    super(fields);
+  }
+
+  private createEmailMessage({
+    message,
+    to,
+    subject,
+    cc,
+    bcc,
+  }: SendMessageParams): string {
+    const emailLines: string[] = [];
+
+    // Format the recipient(s)
+    const formatEmailList = (emails: string | string[]): string =>
+      Array.isArray(emails) ? emails.join(",") : emails;
+
+    emailLines.push(`To: ${formatEmailList(to)}`);
+    if (cc) emailLines.push(`Cc: ${formatEmailList(cc)}`);
+    if (bcc) emailLines.push(`Bcc: ${formatEmailList(bcc)}`);
+    emailLines.push(`Subject: ${subject}`);
+    emailLines.push("");
+    emailLines.push(message);
+
+    // Convert the email message to base64url string
+    const email = emailLines.join("\r\n").trim();
+    // this encode may be an issue
+    return Buffer.from(email).toString("base64url");
+  }
+
+  protected async _call({
+    message,
+    to,
+    subject,
+    cc,
+    bcc,
+  }: SendMessageParams): Promise<string> {
+    return this.sendMessage({ message, to, subject, cc, bcc });
```

Comment:
Should this be in a separate function? Can't the logic from `sendMessage` be inside this `_call` method?

Reply:
I think it can be in _call if needed.